plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,63 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostConnectabilityStatus returns a promise that resolves with `true` if
+// the provided netaddress is connectable, or `false` if it is not connectable.
+// The input 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostConnectabilityStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
+	socket.on('error', () => resolve(false))
+	socket.on('connect', () => {
+		socket.destroy()
+		resolve(true)
+	})
+})
+
+// hostWorkingStatusSaga compares the host's current `settingscalls` metric and
+// sets `working` to false if it has not increased.
+function *hostWorkingStatusSaga() {
+	try {
+		const hostData = yield siadCall('/host')
+		const nPreviousSettingsCalls = yield select((state) => state.hostingReducer.get('nSettingsCalls'))
+		if (nPreviousSettingsCalls >= hostData.networkmetrics.settingscalls) {
```

Comment:
I would look, over the course of 15 minutes, for at least 3 settings calls, instead of just 1